tonilopezmr/day5/MazeOfJumpInstructions.kt

```diff
@@ -0,0 +1,61 @@
+package day5
+
+import org.junit.Assert.*
+import org.junit.Test
+import java.io.File
+import java.util.*
+
+class MazeOfJumpInstructions {
+
+  private fun jumpInstI(input: List<Int>): Int {
+    var offset = 0
+    var steps = 0
+    val instructions = input.toMutableList()
+
+    while (offset < input.size) {
+      val aux = instructions[offset]
+      instructions.removeAt(offset)
+      instructions.add(offset, aux + 1)
+      offset += aux
+      steps++
+    }
+
+    return steps
+  }
+
+  private fun jumpInst(input: List<Int>): Int = jumpInst(0, 0, input)
+
+  tailrec private fun jumpInst(offset: Int, step: Int, input: List<Int>): Int {
+    return if (offset >= input.size) step
+    else jumpInst(offset + input[offset], step + 1, input.take(offset) + (input[offset] + 1) + input.takeLast(input.size - (offset + 1)))
+  }
```

Comment:
This recursion is way lighter than mine. It is logic it takes less half the time 💯

Reply:
Lol